### PR TITLE
48: Filtering Sorting for listings

### LINKED ISSUE #48: Filtering+Sorting for listings

### PR DESCRIPTION
**Pull request by GolanHassin**
## Describe your changes
A filtering and sorting functionality was added to the seller listing and all listing pages. 
When a user visits these pages there is a drop-down that allows them to sort the listings by:
Alphabetic title, Reverse alphabetic title, Alphabetic author, Most expensive to cheapest, and
Cheapest to most expensive. This was done by adding a sorting method to the ListingGrid.tsx
file and updating the states whenever an option to in the MenuItems is selected

## Issue ticket number and link
https://github.com/SYSC4806-Group13/SYSC4806-Project/issues/48

## How Has This Been Tested?
Self-testing was done and it was confirmed after running the program.
Different listings were created and then sorted with the drop-down.

### CODE REVIEW

File: frontend/src/components/Listing/ListingGrid.tsx [hunk 1]
```diff
@@ -22,17 +22,61 @@ export interface IListingGridProps {
...
     if (props.listings.length === 0) {
         return (
             <Typography variant="h4" align='center' color='text.secondary' mt={2}>
                 No listings found
             </Typography>
         );
     }
+
+    const sortListings = (listings : Array<any>) => {
+        console.log("I was called");
```
Comment (NosBos) [inline frontend/src/components/Listing/ListingGrid.tsx]: Remove console.log

File: frontend/src/components/Listing/ListingGrid.tsx [hunk 2]
```diff
@@ -22,17 +22,61 @@ export interface IListingGridProps {
...
+        const myClonedArray: any[] = [];
+        listings.forEach(val => myClonedArray.push(Object.assign({}, val)));
+        switch (sortingType){
+            case "A-Z":
```
Comment (NosBos) [inline frontend/src/components/Listing/ListingGrid.tsx]: maybe replace A-Z stuff with a name that correctly implies that its sorting by title

File: frontend/src/components/Listing/ListingGrid.tsx (hunk 1 above)
Comment (GolanHassin) [reply frontend/src/components/Listing/ListingGrid.tsx]: Removed

File: frontend/src/components/Listing/ListingGrid.tsx (hunk 2 above)
Comment (GolanHassin) [reply frontend/src/components/Listing/ListingGrid.tsx]: Names replaced with more specific titles

**Comment by Justgo13** [comment]
Reverse alphabetical not working

**Review by Justgo13 — APPROVED** [review]

**Pull request merged**